### PR TITLE
Allow cancellation during extract method's symbol walking

### PR DESCRIPTION
**Pull request by amcasey**
...since it's potentially expensive.

### CODE REVIEW

File: src/compiler/symbolWalker.ts [hunk 1]
```diff
@@ -158,6 +162,10 @@ namespace ts {
             }
 
             function visitSymbol(symbol: Symbol): boolean {
+                if (cancellationToken) {
```
Comment (minestarks) [inline src/compiler/symbolWalker.ts]: I have 2 fairly ignorant questions
- Are cancellation checks expensive (I seem to recall that it's based on checking a named pipe, which sounds scary)? Is it throttled?
- How many times do we expect `visitSymbol` to be called in a "typical" operation?

I guess the overall question I'm asking is whether you're fairly sure this check isn't a performance burden.

Comment (vladima) [reply src/compiler/symbolWalker.ts]: [it is](https://github.com/Microsoft/TypeScript/blob/54dfdd77fb222be0cf6aa16a9e962b1d2f9a8e23/src/server/lsHost.ts#L19)

Comment (billti) [reply src/compiler/symbolWalker.ts]: FYI: Some crude overhead measurements from back when I prototyped this: https://github.com/billti/cancellation . Definitely use the throttling mechanism. We've seen significant perf regressions when this wasn't done in prior cancellation checks.

Comment (amcasey) [reply src/compiler/symbolWalker.ts]: As Vlad said, cancellation checks are throttled.  Having said that, I don't have a sense of how expensive cancellability is.

File: src/services/refactors/extractMethod.ts [hunk 2]
```diff
@@ -932,7 +933,8 @@ namespace ts.refactor.extractMethod {
         scopes: Scope[],
         enclosingTextRange: TextRange,
         sourceFile: SourceFile,
-        checker: TypeChecker) {
+        checker: TypeChecker,
+        cancellationToken: CancellationToken) {
```
Comment (weswigham) [inline src/services/refactors/extractMethod.ts]: With the other changes, I don't think this argument is used anywhere anymore.

**Comment by amcasey** [comment]
Per @weswigham's suggestion, handle cancellation in the Accept predicate, rather than adding a new SymbolWalker API.

File: src/services/refactors/extractMethod.ts (hunk 2 above)
Comment (amcasey) [reply src/services/refactors/extractMethod.ts]: If you mean `cancellationToken`, I believe it is closed over in the accept lambda below.

Comment (weswigham) [reply src/services/refactors/extractMethod.ts]: @amcasey Ack, sorry, thought I was reading a different file here. Nevermind. All good.

**Pull request merged**